pkg/reconciler/revision/background.go

```diff
@@ -170,35 +171,31 @@ func (r *backgroundResolver) addWorkItems(rev *v1.Revision, name types.Namespace
 		registriesToSkip: registriesToSkip,
 		statuses:         make([]v1.ContainerStatus, len(rev.Spec.Containers)),
 		remaining:        len(rev.Spec.Containers),
+		workItems:        make([]workItem, len(rev.Spec.Containers)),
 		completionCallback: func() {
 			r.enqueue(name)
 		},
 	}
 
 	for i := range rev.Spec.Containers {
-		image := rev.Spec.Containers[i].Image
-
-		r.queue.AddRateLimited(workItem{
+		item := workItem{
 			revision: name,
 			timeout:  timeout,
 			name:     rev.Spec.Containers[i].Name,
-			image:    image,
+			image:    rev.Spec.Containers[i].Image,
```

Comment:
nit: Not sure if the compiler optimizes this but... Is it worth caching `rev.Spec.Containers[i]`?

Reply:
I doubt it makes any real difference benchmark-wise, but actually we can do this in the range statement so might as well 👍